content/learn/book/migration-guides/0.7-0.8/_index.md

```diff
@@ -0,0 +1,511 @@
+       log!("found visible entity");
+    }
+  }
+}
+```
+
+### [SpatialBundle](https://github.com/bevyengine/bevy/pull/5344)
+
+If you previously needed a `TransformBundle` and a `VisibilityBundle` you should now use a `SpatialBundle`.
+
+If you were defining your own bundle with `Transform` and `Visibility` components, make sure they now have a `ComputedVisiblity` component.
+
+### [Use Affine3A for GlobalTransform to allow any affine transformation](https://github.com/bevyengine/bevy/pull/4379)
+
+`GlobalTransform` fields have changed
+
+* Replace `global_transform.translation` by `global_transform.translation()` (For other fields, use the `compute_transform` method)
+* `GlobalTransform` do not support non-linear scales anymore, we'd like to hear from you if it is an inconvenience for you
+* If you need the `scale`, `rotation` or `translation` property you can now use `global_transform.to_scale_rotation_translation()`
+
+```rust
+// 0.7
+let transform = Transform::from(*global_transform);
+transform.scale
+transform.rotation
+transform.translation
+
+// 0.8
+let (scale, rotation, translation) = global_transform.to_scale_rotation_translation();
+```
+
+### [add a SceneBundle to spawn a scene](https://github.com/bevyengine/bevy/pull/2424)
+
+```rust
+// 0.7
+commands.spawn_scene(asset_server.load("models/FlightHelmet/FlightHelmet.gltf#Scene0"));
+
+//0.8
+commands.spawn_bundle(SceneBundle {
+    scene: asset_server.load("models/FlightHelmet/FlightHelmet.gltf#Scene0"),
+    ..Default::default()
+});
+```
+
+The scene will be spawned as a child of the entity with the `SceneBundle`
+
+### [Make ScalingMode more flexible](https://github.com/bevyengine/bevy/pull/3253)
+
+Adds ability to specify scaling factor for `WindowSize`, size of the fixed axis for `FixedVertical` and `FixedHorizontal` and a new `ScalingMode` that is a mix of `FixedVertical` and `FixedHorizontal`
+
+### [Allow closing windows at runtime](https://github.com/bevyengine/bevy/pull/3575)
+
+`bevy::input::system::exit_on_esc_system` has been removed. Use `bevy::window::close_on_esc` instead.
+
+`CloseWindow` has been removed. Use `Window::close` instead.
+The Close variant has been added to `WindowCommand`. Handle this by closing the relevant window.
+
+### [Make RunOnce a non-manual System impl](https://github.com/bevyengine/bevy/pull/3922)
+
+The run criterion `RunOnce`, which would make the controlled systems run only once, has been replaced with a new run criterion function ShouldRun::once. Replace all instances of RunOnce with `ShouldRun::once`.
+
+### [Move system_param fetch struct into anonymous scope to avoid name collisions](https://github.com/bevyengine/bevy/pull/4100)
+
+For code that was using a system param's fetch struct, such as EventReader's EventReaderState, the fetch struct can now be identified via the SystemParam trait associated type Fetch, e.g. for `EventReader<T>` it can be identified as `<EventReader<'static, 'static, T>` as SystemParam>::Fetch
```

Comment:
```suggestion
For code that was using a system param's fetch struct, such as EventReader's EventReaderState, the fetch struct can now be identified via the SystemParam trait associated type Fetch, e.g. for `EventReader<T>` it can be identified as `<EventReader<'static, 'static, T> as SystemParam>::Fetch`
```